src/main/java/lithium/io/ews/EwsWriter.java

```diff
@@ -16,281 +16,551 @@
-			}
-			else if ( content instanceof BinaryContent )
-			{
-				final BinaryContent binaryContent = (BinaryContent)content;
-				final byte[] bytes = binaryContent.getBytes();
-				cumulativeContentLength += bytes.length + 4;
-			}
-			else if ( content != null )
-			{
-				throw new IllegalArgumentException( "Unsupported content: " + content );
-			}
-
-		}
-	}
-
-	private void writeZeroes( final int count )
-	throws IOException
-	{
-		for ( int i = 0; i < count; i++ )
-		{
-			_out.write( 0 );
-		}
-	}
-
-	private void writeString( final String string )
-	throws IOException
-	{
-		_out.write( string.getBytes( getCharset() ) );
-	}
-
-	private void writeTimestamp( final Date date )
-	throws IOException
-	{
-		if ( date == null )
-		{
-			writeDouble( 0.0 );
-		}
-		else
-		{
-			final Calendar calendar = Calendar.getInstance();
-
-			calendar.setTime( date );
-			calendar.set( Calendar.HOUR_OF_DAY, 0 );
-			calendar.set( Calendar.MINUTE, 0 );
-			calendar.set( Calendar.SECOND, 0 );
-			calendar.set( Calendar.MILLISECOND, 0 );
-			final Date startOfDay = calendar.getTime();
-
-			// It appears that 1 January 1900 does not compute.
-			calendar.set( 1899, Calendar.DECEMBER, 30, 0, 0, 0 );
-			final Date epoch = calendar.getTime();
-
-			// TODO: Not accurate for various reasons. Use Joda-Time.
-			final long dateDifference = date.getTime() - epoch.getTime();
-			final long timeDifference = date.getTime() - startOfDay.getTime();
-			final double timestamp = (double)( dateDifference / 86400000L ) + (double)timeDifference / 86400000.0;
-
-			writeDouble( timestamp );
-		}
-	}
-
-	private void writeDouble( final double d )
-	throws IOException
-	{
-		writeLong( Double.doubleToLongBits( d ) );
-	}
-
-	private void writeLong( final long i )
-	throws IOException
-	{
-		writeInt( (int)i );
-		writeInt( (int)( i >> 32 ) );
-	}
-
-	private void writeInt( final int i )
-	throws IOException
-	{
-		_out.write( i & 0xff );
-		_out.write( ( i >> 8 ) & 0xff );
-		_out.write( ( i >> 16 ) & 0xff );
-		_out.write( ( i >> 24 ) & 0xff );
-	}
-
-	private void writeShort( final int i )
-	throws IOException
-	{
-		_out.write( i & 0xff );
-		_out.write( ( i >> 8 ) & 0xff );
-	}
-
-	private void writePaddedString( final String string, final int length )
-	throws IOException
-	{
-		if ( string == null )
-		{
-			writeZeroes( length );
-		}
-		else
-		{
-			final byte[] bytes = string.getBytes( getCharset() );
-			_out.write( bytes );
-			writeZeroes( length - bytes.length );
-		}
-	}
-
-	public void close()
-	throws IOException
-	{
-		_out.close();
-	}
+public class EwsWriter {
+    private static final int SCHEDULE_ENTRY_LENGTH = 1816;
+
+    private final OutputStream _out;
+
+    private Charset _charset = Charset.forName(Config.charset);
+
+    public EwsWriter(final OutputStream out) {
+        _out = out;
+    }
+
+    public EwsWriter(final File file) throws IOException {
+        file.createNewFile();
+        _out = new FileOutputStream(file);
```

Comment:
Stream is not closed. File constructor is an anti pattern (unless the writer is AutoClosable). Open and close the stream in the calling code in a try-with-resources block.